internal/pkg/term/prompt/select_test.go

```diff
@@ -0,0 +1,170 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package prompt
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOption_String(t *testing.T) {
+	testCases := map[string]struct {
+		input  Option
+		wanted string
+	}{
+		"should render the value with a tab if there is no hint": {
+			input: Option{
+				Value: "Help me decide!",
+			},
+			wanted: "Help me decide!\t",
+		},
+		"should render a hint in parenthesis separated by a tab": {
+			input: Option{
+				Value: "Load Balanced Web Service",
+				Hint:  "ELB -> ECS on Fargate",
+			},
+			wanted: "Load Balanced Web Service\t(ELB -> ECS on Fargate)",
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			require.Equal(t, tc.wanted, tc.input.String())
+		})
+	}
+}
+
+func TestPrompt_SelectOption(t *testing.T) {
+	t.Run("should return ErrEmptyOptions if there are no options", func(t *testing.T) {
+		_, err := New().SelectOption("to be or not to be?", nil)
+		require.EqualError(t, err, ErrEmptyOptions.Error())
+	})
+	t.Run("should return value without hint", func(t *testing.T) {
+		// GIVEN
+		var p Prompt = func(p survey.Prompt, out interface{}, _ ...survey.AskOpt) error {
+			sel := p.(*prompt).prompter.(*survey.Select)
+			require.ElementsMatch(t, []string{
+				"Load Balanced Web Service  (ELB -> ECS on Fargate)",
+				"Backend Service            (ECS on Fargate)",
+				"Scheduled Job              (CW Event -> StateMachine -> Fargate)",
+			}, sel.Options)
+			result := out.(*string)
+			*result = "Load Balanced Web Service     (ELB -> ECS on Fargate)"
+			return nil
+		}
+		opts := []Option{
+			{
+				Value: "Load Balanced Web Service",
+				Hint:  "ELB -> ECS on Fargate",
+			},
+			{
+				Value: "Backend Service",
+				Hint:  "ECS on Fargate",
+			},
+			{
+				Value: "Scheduled Job",
+				Hint:  "CW Event -> StateMachine -> Fargate",
+			},
+		}
+
+		// WHEN
+		actual, err := p.SelectOption("Which workload type?", opts)
+
+		// THEN
+		require.NoError(t, err)
+		require.Equal(t, "Load Balanced Web Service", actual)
+	})
+	t.Run("should return value without extra spaces when there are no hints", func(t *testing.T) {
+		// GIVEN
+		var p Prompt = func(p survey.Prompt, out interface{}, _ ...survey.AskOpt) error {
+			sel := p.(*prompt).prompter.(*survey.Select)
+			require.ElementsMatch(t, []string{
+				"Load Balanced Web Service  (ELB -> ECS on Fargate)",
+				"Help me decide!",
```

Comment:
ohhh dumb question: I wonder where along the line did the `\t` after `"Help me decide!"` go way?

Reply:
It's a good question haha, it's just the way [tabwriter](https://golang.org/pkg/text/tabwriter/) works. If we don't put `\t` for each line then the calculation for the spacing becomes all wrong. For example instead of seeing:
```
Load Balanced Web Service  (ELB -> ECS on Fargate)
Help me decide!
Backend Service            (ECS on Fargate)
```
Users will see:
```
Load Balanced Web Service  (ELB -> ECS on Fargate)
Help me decide!
Backend Service  (ECS on Fargate)
```
Updated the test to capture the scenario above https://github.com/aws/copilot-cli/pull/2092/commits/8aab7fe7ced14aa39ae30d3fb26d5becc59442ff! It also caught a bug with the extra space in case of no hint!

Reply:
Nicee!! 8aa7fe answered all my questions! Thanks for addressing that :shipit: